api/src/integrations/trav-com-integration/serializers/accounts-receivable-invoice-details/index-serializer.ts

```diff
@@ -0,0 +1,284 @@
+import { isEmpty, isNil, isUndefined, last, pick, sortBy } from "lodash"
+
+import guessNameCapitalization from "@/integrations/trav-com-integration/utils/guess-name-capitalization"
+import {
+  AccountsReceivableInvoice,
+  AccountsReceivableInvoiceDetail,
+  Segment,
+} from "@/integrations/trav-com-integration/models"
+import { type AccountsReceivableInvoiceDetailWithFlightReconciliation } from "@/integrations/trav-com-integration/services/accounts-receivable-invoice-details/index-service"
+import BaseSerializer from "@/serializers/base-serializer"
+import { FlightReconciliation } from "@/models"
+
+export type AccountsReceivableInvoiceDetailIndexView = Pick<
+  AccountsReceivableInvoiceDetail,
+  | "id"
+  | "invoiceId"
+  | "transactionType"
+  | "vendorNumber"
+  | "vendorName"
+  | "productCode"
+  | "passengerName"
+  | "ticketNumber"
+  | "publishedFare"
+  | "sellingFare"
+  | "referenceFare"
+  | "lowFare"
+  | "tax1"
+  | "grossAmount"
+  | "commissionAmount"
+  | "vatOnCommission"
+  | "freeFieldA"
+  | "travelDate"
+  | "returnDate"
+  | "numberOfDays"
+  | "cityCode"
+  | "profileNumber"
+  | "addedBy"
+> & {
+  // magic attributes
+  agentName: string
+  flightInfo: string
+  finalDestination: string
+  department: string
+  travelerFirstName: string
+  travelerLastName: string
+
+  // associations
+  // TODO: move invoice type definition to accounts-receivable-invoice show serializer
+  invoice: Pick<
+    AccountsReceivableInvoice,
+    | "id"
+    | "invoiceNumber"
+    | "profileNumber"
+    | "profileName"
+    | "department"
+    | "bookingDate"
+    | "systemDate"
+    | "description"
+    | "invoiceRemarks"
+  >
+  // TODO: move segment type definition to segments show serializer
+  segments: Pick<
+    Segment,
+    | "id"
+    | "invoiceId"
+    | "invoiceDetailId"
+    | "legNumber"
+    | "departureCityCode"
+    | "departureInfo"
+    | "arrivalCityCode"
+    | "arrivalInfo"
+    | "airlineCode"
+    | "flightNumber"
+    | "classOfService"
+    | "fareBasis"
+  >[]
+  flightReconciliation: FlightReconciliation | null
+}
+
+export class IndexSerializer extends BaseSerializer<AccountsReceivableInvoiceDetailWithFlightReconciliation> {
```

Comment:
Should I simplify so its just returning the data I want for sync, and/or move it to the `api/src/services/flight-reconciliations/sync-service.ts`?

Reply:
I simplified the returned data.